src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -1369,7 +1369,7 @@ bool ReplicatedMergeTreeQueue::shouldExecuteLogEntry(
             if (!disks.empty() && only_s3_storage && storage.checkZeroCopyLockExists(entry.new_part_name, disks[0], replica_to_execute_merge))
             {
                 constexpr auto fmt_string = "Not executing merge/mutation for the part {}, waiting for {} to execute it and will fetch after.";
-                out_postpone_reason = fmt::format(fmt_string, entry.new_part_name, replica_to_execute_merge);
+                LOG_TEST(LogToStr(out_postpone_reason, log), fmt_string, entry.new_part_name, replica_to_execute_merge);
```

Comment:
I may be wrong about how we use `LogToStr`, but it looks like `out_postpone_reason` is filled only if the message is logged. This means that `out_postpone_reason` will be empty if the current logs level > test.

I may be wrong.